src/algebra/big_operators/basic.lean

```diff
@@ -225,6 +225,15 @@ lemma prod_union [decidable_eq α] (h : disjoint s₁ s₂) :
   (∏ x in (s₁ ∪ s₂), f x) = (∏ x in s₁, f x) * (∏ x in s₂, f x) :=
 by rw [←prod_union_inter, (disjoint_iff_inter_eq_empty.mp h)]; exact (mul_one _).symm
 
+@[to_additive]
+lemma prod_coe [decidable_eq α] :
+  ∏ (i : (s : set α)), f i = ∏ i in s, f i :=
+begin
+  rw [←prod_image, prod_congr _ (λ _ _, rfl)],
+  { ext, simp },
+  { simp },
+end
```

Comment:
Isn't this just a not-very-special case of `prod_subtype`?

Reply:
It might be, but I can't get a clean proof of it using any of the variants of `prod_subtype` in `big_operators.basic`. Do you have an improved proof?

Reply:
Sure,
```suggestion
@[to_additive]
lemma prod_coe [decidable_eq α] :
  ∏ (i : (s : set α)), f i = ∏ i in s, f i :=
(finset.prod_subtype s (λ _, iff.rfl) f).symm
```
although `finset.prod_subtype` is in `data/fintype/card`. Perhaps it should be moved?

Reply:
Thanks, changed!

Reply:
That won't work because `finset.prod_subtype` isn't yet defined here!

Reply:
Ah. Are you suggesting then to move `finset.prod_subtype` and use that instead?

Reply:
EIther move `finset.prod_subtype` here, or move your lemma there. I think your lemma is probably worth having, expecially since it looks like a good simp lemma.

Reply:
Amended, thanks. I moved this lemma lower so that it can be in the `fintype` namespace rather than the finset one.